src/client.rs

```diff
@@ -84,6 +84,37 @@ impl Client {
             .await
     }
 
+    /// Patches subresource status {} in a given Resource.
+    /// Subresource status {} must be defined beforehand in the Crd.
+    /// There exist four different patch strategies (Merge currently used for apply_patch_status):
+    /// 1) Apply (https://kubernetes.io/docs/reference/using-api/api-concepts/#server-side-apply)
+    ///   Starting from Kubernetes v1.18, you can enable the Server Side Apply feature so that the control plane tracks managed fields for all newly created objects.
+    /// 2) Json (https://tools.ietf.org/html/rfc6902):
+    ///   This is supported on crate feature jsonpatch only
+    /// 3) Merge (https://tools.ietf.org/html/rfc7386):
+    ///   For example, if you want to update a list you have to specify the complete list and update everything  
+    /// 4) Strategic
+    ///   With a strategic merge patch, a list is either replaced or merged depending on its patch strategy.
+    ///   The patch strategy is specified by the value of the patchStrategy key in a field tag in the Kubernetes source code.
+    ///   For example, the Containers field of PodSpec struct has a patchStrategy of merge.
+    pub async fn apply_patch_status<T, S>(&self, resource: &T, status: &S) -> OperatorResult<T>
+    where
+        T: Clone + DeserializeOwned + Meta + Resource,
+        S: Serialize,
+    {
+        let new_status = Patch::Merge(serde_json::json!({
+            "apiVersion": T::API_VERSION,
```

Comment:
Is this really necessary? 
The docs of the `patch_status` method doesn't have it.

Reply:
You mean apiVersion? Yes. With spark operator:
source: KubeError { source: Api(ErrorResponse { status: "Failure", message: "Incorrect version specified in apply patch. Specified patch version: , expected: spark.stackable.tech/v1", reason: "BadRequest", code: 400